### ISSUE TITLE
Docker usage

### ISSUE THREAD

**Issue by uguraltinsoy** [liamcottle/meshtastic-map#69]
How can I connect to my own mqtt server when using with Docker

**Comment by KomelT** [comment]
You must specify options from readme in MQTT_OPTS ENV variable in docker-compose.yml

`...
  meshtastic-mqtt:
    container_name: meshtastic-mqtt
    depends_on:
      database:
        condition: service_healthy
    build:
      context: .
      dockerfile: ./Dockerfile
    command: /app/docker/mqtt.sh
    environment:
      DATABASE_URL: "mysql://root:password@database:3306/meshtastic-map?connection_limit=100"
      MQTT_OPTS: "--mqtt-broker-url 'your-mqtt.tld'" # add any custom mqtt.js options here`
...

**Comment by KomelT** [comment]
@uguraltinsoy was answer ok? If yes please close the issue.

**Comment by gally789** [comment]
This thread was helpful - however, I tried the above but got an error.  It looks like the missing protocol suggests the mqtt:/ prefix is not there within the command.

meshtastic-mqtt  | 
meshtastic-mqtt  | 
meshtastic-mqtt  | Starting mqtt listener
meshtastic-mqtt  | /app/node_modules/mqtt/build/lib/connect/index.js:66
meshtastic-mqtt  |             throw new Error('Missing protocol');
meshtastic-mqtt  |             ^
meshtastic-mqtt  | 
meshtastic-mqtt  | Error: Missing protocol
meshtastic-mqtt  |     at Object.connect (/app/node_modules/mqtt/build/lib/connect/index.js:66:19)
meshtastic-mqtt  |     at Object.<anonymous> (/app/src/mqtt.js:238:21)
meshtastic-mqtt  |     at Module._compile (node:internal/modules/cjs/loader:1469:14)
meshtastic-mqtt  |     at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
meshtastic-mqtt  |     at Module.load (node:internal/modules/cjs/loader:1288:32)
meshtastic-mqtt  |     at Module._load (node:internal/modules/cjs/loader:1104:12)
meshtastic-mqtt  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
meshtastic-mqtt  |     at node:internal/main/run_main_module:28:49
meshtastic-mqtt  | 
meshtastic-mqtt  | Node.js v20.18.0
meshtastic-mqtt exited with code 1

My docker compose looks like this...
  meshtastic-mqtt:
    container_name: meshtastic-mqtt
    depends_on:
      database:
        condition: service_healthy
    build:
      context: .
      dockerfile: ./Dockerfile
    command: /app/docker/mqtt.sh
    environment:
      DATABASE_URL: "mysql://root:berks!mqtt@database:3306/meshtastic-map?connection_limit=100"
      #MQTT_OPTS: "" # add any custom mqtt.js options here
      MQTT_OPTS: "--mqtt-broker-url 'mqtt://10.10.100.253' --mqtt-username 'username_here' --mqtt-password 'password_here'"

Can you help identify the issue?

**Comment by liamcottle** [comment]
I'd say try removing the single quotes around your option values.

**Comment by gally789** [comment]
> I'd say try removing the single quotes around your option values.

Thanks, that is working.  👍 

For future ref my example config is:
MQTT_OPTS: "--mqtt-broker-url mqtt:servername --mqtt-username username--mqtt-password password --mqtt-topic /msh/EU868 --collect-service-envelopes --collect-text-messages --collect-waypoints --collect-neighbour-info --collect-map-reports"

Is there something wrong with the docker package?  
I can see meshtastic mqtt data coming into my server.  I also see the mqtt.js listener is connected.  I see the database populated with meshtastic tables in phpMyAdmin.  I do not see any data in any of the tables and there are the following error messages from the database container...

2024-10-26  7:59:51 5 [Warning] Aborted connection 5 to db: 'prisma_migrate_shadow_db_f75a8b2f-2fb3-4b93-80b4-42a6c6a47959' user: 'root' host: '172.18.0.4' (Got an error reading communication packets)

2024-10-26  7:59:52 6 [Warning] Aborted connection 6 to db: 'prisma_migrate_shadow_db_589ae7fe-8c23-4355-a91e-107e09f39783' user: 'root' host: '172.18.0.4' (Got an error reading communication packets)

2024-10-26  7:59:53 7 [Warning] Aborted connection 7 to db: 'prisma_migrate_shadow_db_f4e98cdd-9c69-49fd-9305-37d118a99fad' user: 'root' host: '172.18.0.4' (Got an error reading communication packets)

2024-10-26  7:59:54 8 [Warning] Aborted connection 8 to db: 'prisma_migrate_shadow_db_ed4eadb9-d9f2-4ad8-9634-2e2f4b575cf0' user: 'root' host: '172.18.0.4' (Got an error reading communication packets)

2024-10-26  7:59:54 4 [Warning] Aborted connection 4 to db: 'meshtastic-map' user: 'root' host: '172.18.0.4' (Got an error reading communication packets)

**Comment by gally789** [comment]
Could it be because I do not have a base64 decode option?
--decryption-keys <base64DecryptionKey>

Also really struggling to wrap my head around this.  How would I add a custom channel?  Do I paste the PSK from the channel as the decryption key?

**Comment by gally789** [comment]
I've rebuilt this docker image several times and had no success.  I think there is an issue with connections to the database.  The errors are still as above. 

Has anyone successfully built this in docker?

**Comment by gally789** [comment]
I resolved my issue.  MQTT params are as follows...

MQTT_OPTS: "--mqtt-broker-url mqtt://servername --mqtt-username username --mqtt-password password  --collect-service-envelopes --collect-text-messages --collect-waypoints --collect-neighbour-info --collect-map-reports"

The issue was my mqtt topic was not working.  I tried "--mqtt-topic /msh/EU_868"  and "--mqtt-topic /msh/EU_868/#" but nothing would work. I removed this option and went with the default and it's fine.

I still have not worked out how to add a custom channel but I am sure I will get there.  Help would be appreciated.

I think this issue/question can be closed.
